### ISSUE TITLE
MAS build fails if I include sandbox bool in entitlement files 

### ISSUE THREAD

**Issue by MartinBarker** [electron/electron#27679]
Hello, I have made an electron 6.7.0 app called Digify and I have it on a branch "mas-attempt-after-redesign" here : https://github.com/MartinBarker/digify/tree/mas-attempt-after-redesign

My app uses ffmpeg and I am trying to publish it to the Mac Apple Store, so I made sure to statically build ffmpeg so that It can be sandboxed (details on how to setup/run my app are in the README file)

Right now If I run `npm run build-mas && sh mas-sign-script.sh ` it will build the app for the Mac Apple Store (mas) and then run a script to sign the outputted mas file. After running the above command I can use the outputted Digify.app fine:
![Screen Shot 2021-02-08 at 10 05 24 PM](https://user-images.githubusercontent.com/27025504/107322702-d00bcd00-6a59-11eb-84b4-663507e400e8.png)

The .sh script signs and outputs the file as Digify-mac.pkg in my projects folder, If I try to submit it using the mac transporter app, I get an issue saying app sandbox is not enabled:
![Screen Shot 2021-02-08 at 10 11 13 PM](https://user-images.githubusercontent.com/27025504/107323160-a606da80-6a5a-11eb-9461-ab60552506e0.png)
Full issue text:
```
ERROR ITMS-90296: "App sandbox not enabled. The following executables must include the "com.apple.security.app-sandbox" entitlement with a Boolean value of true in the entitlements property list: [( "com.martinbarker.digifyunique.pkg/Payload/Digify.app/Contents/Frameworks/Digify Helper (GPU).app/Contents/MacOS/Digify Helper (GPU)", "com.martinbarker.digifyunique.pkg/Payload/Digify.app/Contents/Frameworks/Digify Helper (Plugin).app/Contents/MacOS/Digify Helper (Plugin)", "com.martinbarker.digifyunique.pkg/Payload/Digify.app/Contents/Frameworks/Digify Helper (Renderer).app/Contents/MacOS/Digify Helper (Renderer)", "com.martinbarker.digifyunique.pkg/Payload/Digify.app/Contents/Frameworks/Digify Helper.app/Contents/MacOS/Digify Helper", "com.martinbarker.digifyunique.pkg/Payload/Digify.app/Contents/Library/LoginItems/Digify Login Helper.app/Contents/MacOS/Digify Login Helper", "com.martinbarker.digifyunique.pkg/Payload/Digify.app/Contents/MacOS/Digify", "com.martinbarker.digifyunique.pkg/Payload/Digify.app/Contents/Resources/app.asar.unpacked/node_modules/trash/lib/macos-trash", "com.martinbarker.digifyunique.pkg/Payload/Digify.app/Contents/Resources/ffmpeg", "com.martinbarker.digifyunique.pkg/Payload/Digify.app/Contents/Resources/ffprobe" )] Refer to App Sandbox page at https://developer.apple.com/documentation/security/app_sandbox for more information on sandboxing your app."
```

My package.json build section for MAS looks like this:
```
"mas": {
      "artifactName": "${productName}-${version}-${arch}.${ext}",
      "hardenedRuntime": false,
      "entitlements": "entitlements.mas.plist",
      "entitlementsInherit": "entitlements.mas.inherit.plist",
      "provisioningProfile": "embedded.provisionprofile",
      "binaries": [
        "dist/mas/Digify.app/Contents/Resources/ffmpeg",
        "dist/mas/Digify.app/Contents/Resources/ffprobe",
        "dist/mas/Digify.app/Contents/Resources/app.asar.unpacked/node_modules/trash/lib/macos-trash"
      ]
    },
```

if I add the sandbox code to my entitlements.mas.plist file so it looks like this:
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>com.apple.security.app-sandbox</key>
	<true/>
	<key>com.apple.security.application-groups</key>
	<array>
		<string>LV6WXG529F.com.martinbarker.digifyunique</string>
	</array>
	<key>com.apple.security.network.client</key>
	<true/>
	<key>com.apple.security.files.user-selected.read-write</key>
	<true/>
</dict>
</plist>
```
and also add the sandbox text to my entitlements.mas.inherit.plist file:
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>com.apple.security.app-sandbox</key>
	<true/>
	<key>com.apple.security.inherit</key>
	<true/>
</dict>
</plist>

``` 

And then run the command to build and sign my app again: `npm run build-mas && sh mas-sign-script.sh `

Once it is finished, if I try to open the finished Digify mas build file that worked previously, it now crashes with an error saying "code signature invalid".

While the mas app file crashes, the Digify-mac.pkg file that the .sh file signed can now be submitted on transporter fine without the sandbox issue from before:
![Screen Shot 2021-02-08 at 10 32 20 PM](https://user-images.githubusercontent.com/27025504/107324815-891fd680-6a5d-11eb-8890-03d9d4f0e172.png)

So In summary my issue is this:

If I remove the '<key>com.apple.security.app-sandbox</key><true/>' line from my mas entitlements files, the built mas app works, but transporter rejects my submission because I am missing the sandbox lines.

If I add the '<key>com.apple.security.app-sandbox</key><true/>' lines into my mas entitlements files, the built mas now crashes with a code signature invalid error, but transporter will accept it (but it is accepting a broken app file that wont open).

Is there some sandbox step that I'm missing? I know that packaging ffmpeg in an app on the mac apple store is technically possible because of this application lossless-cut i found that does the same thing: https://github.com/mifi/lossless-cut

But somewhere in my implementation is an issue causing my mas build to crash with a code signing error if I include <sandbox> lines in my entitlement files. Can anyone help me please? Thank you

![Screen Shot 2021-02-08 at 10 18 53 PM](https://user-images.githubusercontent.com/27025504/107323754-aeabe080-6a5b-11eb-85d9-14b16c850846.png)

Full crash text:
```
Process:               Digify [17773]
Path:                  /Users/USER/Documents/*/Digify.app/Contents/MacOS/Digify
Identifier:            com.martinbarker.digifyunique
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           Digify [17773]
User ID:               501

Date/Time:             2021-02-08 22:27:44.682 -0800
OS Version:            Mac OS X 10.14.6 (18G8012)
Report Version:        12
Anonymous UUID:        92BF9AB4-24C8-FB5B-9518-DCEE13532431

Sleep/Wake UUID:       9B8ABFB4-EF56-4DDA-A451-4C6C644FE573

Time Awake Since Boot: 12000 seconds
Time Since Wake:       2200 seconds

System Integrity Protection: disabled

Crashed Thread:        0

Exception Type:        EXC_CRASH (Code Signature Invalid)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace CODESIGNING, Code 0x1

kernel messages:

VM Regions Near 0 (cr2):
--> 
    __TEXT                 000000010aa6f000-000000010aa97000 [  160K] r-x/r-x SM=COW  

Thread 0 Crashed:
0                                 	0x0000000112bdf000 _dyld_start + 0

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000000  rbx: 0x0000000000000000  rcx: 0x0000000000000000  rdx: 0x0000000000000000
  rdi: 0x0000000000000000  rsi: 0x0000000000000000  rbp: 0x0000000000000000  rsp: 0x00007ffee5190ba8
   r8: 0x0000000000000000   r9: 0x0000000000000000  r10: 0x0000000000000000  r11: 0x0000000000000000
  r12: 0x0000000000000000  r13: 0x0000000000000000  r14: 0x0000000000000000  r15: 0x0000000000000000
  rip: 0x0000000112bdf000  rfl: 0x0000000000000200  cr2: 0x0000000000000000
  
Logical CPU:     0
Error Code:      0x00000000
Trap Number:     0


Binary Images:
       0x10aa6f000 -        0x10aa96ff3 + (0) <7C1D13BE-342F-3F53-A359-71D89B99F0AC> 
       0x112bde000 -        0x112c4870f + (655.1.1) <EA5BF483-08FB-31C9-B31E-C0D0FD94FE89> 

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 7290
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=772K resident=0K(0%) swapped_out_or_unallocated=772K(100%)
Writable regions: Total=8400K written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=8400K(100%)
 
                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
STACK GUARD                       56.0M        1 
Stack                             8192K        1 
__DATA                             244K        3 
__LINKEDIT                         184K        2 
__TEXT                             588K        2 
shared memory                        8K        2 
===========                     =======  ======= 
TOTAL                             65.0M       11 

Model: MacBookAir7,2, BootROM 426.0.0.0.0, 2 processors, Intel Core i5, 1.6 GHz, 4 GB, SMC 2.27f2
Graphics: kHW_IntelHDGraphics6000Item, Intel HD Graphics 6000, spdisplays_builtin
Memory Module: BANK 0/DIMM0, 2 GB, DDR3, 1600 MHz, 0x80CE, 0x4B3445384533303445452D45474345000000
Memory Module: BANK 1/DIMM0, 2 GB, DDR3, 1600 MHz, 0x80CE, 0x4B3445384533303445452D45474345000000
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0x117), Broadcom BCM43xx 1.0 (7.77.61.3 AirPortDriverBrcmNIC-1305.11)
Bluetooth: Version 6.0.14d12, 3 services, 18 devices, 1 incoming serial ports
Network Service: Wi-Fi, AirPort, en0
Serial ATA Device: APPLE SSD SM0256G, 251 GB
USB Device: USB 3.0 Bus
USB Device: BRCM20702 Hub
USB Device: Bluetooth USB Host Controller
Thunderbolt Bus: MacBook Air, Apple Inc., 27.2

```

**Comment by ckerr** [comment]
Hi @MartinBarker, thanks for reaching out and reporting this bug.

I appreciate that you clearly put some time into laying out the issue, but Electron 6.7.0 reached the end of its support in [May 2020](https://www.electronjs.org/blog/electron-9-0#end-of-support-for-6xy). Because of the age of that version, it's unlikely that the Electron team will have any resources to  spend on this issue.

If you're still experiencing this issue on a [supported version](https://www.electronjs.org/docs/tutorial/support#supported-versions) of Electron, please file a new issue for that version of Electron.

Thanks in advance! Your help is appreciated.

**Comment by ckerr** [comment]
Wait, there is no Electron 6.7.0 release so I think that part of the report must be in error.

@MartinBarker what version of Electron are you using?

**Comment by MartinBarker** [comment]
you are correct I included the wrong electron version sorry, my real electron version on my local machine is `electron@11.1.0 `

I also found out an additional clue, I submitted the sandboxed app which crashes when I run it locally, and it got rejected by apple because:
```
Your app links against the following non-public framework(s):

• Contents/Resources/ffprobe/_lzma_stream_decoder
• Contents/Resources/ffmpeg/_lzma_stream_decoder
• Contents/Resources/ffprobe/_lzma_end
• Contents/Resources/ffmpeg/_lzma_end
• Contents/Resources/ffprobe/_lzma_code
• Contents/Resources/ffmpeg/_lzma_code
```

**Comment by MartinBarker** [comment]
built ffmpeg without those frameworks and it worked
